### DISCUSSION TITLE
💡Allow Flying machines to go blast trough any blocks (Not just moss blocks, as you can do in the game right now)

### DISCUSSION THREAD

**Discussion by SalmonSnaked** [PurpurMC/Purpur#1329] [Feature Requests]
Today I watched a Mumbo Jumbo video about 68 Minecraft Blocks with secret features.
And in that video, there was a random game mechanic that I thought would be cool if I could make it apply to more blocks than currently in the game.

If you watch the clip you will understand better: https://youtu.be/hZsJkocN8V4?t=363
(Ignore the weird bug that the flying machine disappeared)

So the idea is that I can specify all the blocks (in the purpur config file) that flying machines can blast through. Currently, in the vanilla game, flying machines can go through moss blocks, but it would be cool if I could specify so that they can for example go through leaves, or any block in the game 🙂

‎ 

This was a simple random idea that I decided to mention since I thought that it could have some cool use cases.
For example, who doesn't want to make an automated flying machine that could act as a drill that could mine stone and other ores underground?

And after all, "Purpur is a drop-in replacement for Paper servers designed for configurability and new, fun, exciting gameplay features." 
So this would be a great addition to Purpur 👍

**Comment by 1ByteBit** [comment]
The reason this happens is due to moss blocks being broken by pistons/blocks moved by pistons. So I should warn you that what you're actually suggesting is "Allow the selection of which blocks are broken by pistons"
There's nothing particularly wrong with this, I just figured I should warn you as the original suggestion sounds a lot different then this
